api/src/handlers/utils.rs

```diff
@@ -54,12 +54,22 @@ pub fn get_output(
 	for x in outputs.iter() {
 		let res = chain.is_unspent(x);
 		match res {
-			Ok(output_pos) => {
-				return Ok((
-					Output::new(&commit, output_pos.height, output_pos.position),
-					x.clone(),
-				));
-			}
+			Ok(output_pos) => match chain.get_unspent_output_at(output_pos.position) {
+				Ok(output) => {
+					match OutputPrintable::from_output(&output, chain.clone(), None, true, true) {
```

Comment:
We may want to parameterize the `include_proof` and `include_merkle_proof` here.
In many cases we don't really care about these and this is where the additional cost would be incurred or avoided.

Reply:
Yeah I'll do. Was thinking that the user probably don't really care about whether having the chance or not. But since it is used by the wallet we will certainly need the performance boost herE.

Reply:
Yeah I think it would be fine to default to both `true`. And any client that cares about performance can opt out of these.

Reply:
> just written that out I'm actually inclined to suggest we default to `false` for both of these and force clients to opt-in to this optional data when they know they really want it...
> 
> We should default to the api being fast and lightweight. Particularly with the size of the rangeproof and the cost of generating the (currently not used by _anybody_) Merkle proof.

The problem is we used to return the rangeproof and merkle proof, depending on how this was called. So defaulting to anything other than `true` would be a non-passive API change, and would require a "major" release. So for v2 APIs, we can default to `false`, but for v1, we probably should default to `true`

Reply:
Okay I have added the `include_rp` and new `include_mp` parameters for outputs by id and by height.

Reply:
I'm hesitant to default `include_merkle_proof` to `true`. I think adding it something about in the release note would be fine.

Reply:
> The problem is we used to return the rangeproof and merkle proof

In which api endpoint?
My understanding is `byids` only ever returned minimal `outputs` as described in this PR and `byheight` defaulted to `false` for `include_rp`.

Reply:
By height was always including the `merkle_proof` https://github.com/mimblewimble/grin/pull/3085/files#diff-6f4987d3855712fd885db9713f06652cR149. Rangeproof was always off by default.

Reply:
I mean there was no field `proof` at that point.

Reply:
> > The problem is we used to return the rangeproof and merkle proof
> 
> In which api endpoint?
> My understanding is `byids` only ever returned minimal `outputs` as described in this PR and `byheight` defaulted to `false` for `include_rp`.

You're right about rangeproofs, but it looks like we included merkle_proofs by default for byheight. I'm ok with changing that during a non-major version, since the odds are low that anybody is actually using merkle proofs. But it's not following the standard we voted on, so we should at least make it clear in the release docs, as mentioned.

https://github.com/mimblewimble/grin/blob/master/api/src/handlers/chain_api.rs#L180
https://github.com/mimblewimble/grin/blob/master/api/src/handlers/chain_api.rs#L146

Reply:
However I agree that it must be for 3.0 since we are moving to a completely different structure.